storage/innobase/xtrabackup/src/redo_log.cc

```diff
@@ -54,36 +57,54 @@ Redo_Log_Reader::Redo_Log_Reader() {
 }
 
 bool Redo_Log_Reader::find_start_checkpoint_lsn() {
-  /* Look for the latest checkpoint */
-  Log_checkpoint_location checkpoint;
+  if (log_sys->m_files.ctx().m_files_ruleset == Log_files_ruleset::CURRENT) {
+    /* Look for the latest checkpoint */
+    Log_checkpoint_location checkpoint;
 
-  if (!recv_find_max_checkpoint(*log_sys, checkpoint)) {
-    xb::error() << "recv_find_max_checkpoint() failed.";
-    return (false);
-  }
+    if (!recv_find_max_checkpoint(*log_sys, checkpoint)) {
+      xb::error() << "recv_find_max_checkpoint() failed.";
+      return (false);
+    }
 
-  /* Redo log file header is never encrypted. */
-  Encryption_metadata unused_encryption_metadata;
+    /* Redo log file header is never encrypted. */
+    Encryption_metadata unused_encryption_metadata;
+
+    auto file_handle =
+        Log_file::open(log_sys->m_files_ctx, checkpoint.m_checkpoint_file_id,
+                       Log_file_access_mode::READ_ONLY,
+                       unused_encryption_metadata, Log_file_type::NORMAL);
+    if (!file_handle.is_open()) {
+      xb::error() << "Failed to open redo log file id "
+                  << checkpoint.m_checkpoint_file_id;
+      return false;
+    }
+    log_scanned_lsn = checkpoint.m_checkpoint_lsn;
+    checkpoint_lsn_start = checkpoint.m_checkpoint_lsn;
 
-  Log_files_context log_files_ctx =
-      Log_files_context{srv_log_group_home_dir, Log_files_ruleset::CURRENT};
+    /* Copy the header into log_hdr_buf */
+    file_handle.read(0, LOG_FILE_HDR_SIZE, log_hdr_buf);
+  } else {
+    const auto logfile0 = log_sys->m_files.file(0);
+    auto file_handle = logfile0->open(Log_file_access_mode::READ_ONLY);
+    if (!file_handle.is_open()) {
+      xb::error() << "Failed to open redo log file id 0";
+      return false;
+    }
+    log_pre_8_0_30::Checkpoint_header chkp_header = {};
+    if (!log_pre_8_0_30::recv_find_max_checkpoint(file_handle, chkp_header)) {
+      xb::error() << "recv_find_max_checkpoint() failed.";
+      return (false);
+    }
 
-  auto file_handle =
-      Log_file::open(log_files_ctx, checkpoint.m_checkpoint_file_id,
-                     Log_file_access_mode::READ_ONLY,
-                     unused_encryption_metadata, Log_file_type::NORMAL);
+    os_offset_t current_file_real_offset = chkp_header.m_checkpoint_offset;
```

Comment:
this can be removed and checkpoint_offset_start set to chkp_header.m_checkpoint_offset;